File: javascript/hackerrank/staircase.js
// https://www.hackerrank.com/challenges/staircase/problem

staircase = (n) => {
    if (n == 0) return null;
    const stairMatrix = [];
    for (let row = 0; row < n; row++) {
        let mRow = [];
        for (let col = 0; col < n; col++) {
            const value = (col >=  parseInt((n-row) - 1 )) ? '#' : ' ';
            mRow.push(value);
        }
        stairMatrix.push(mRow);
    }        
    return stairMatrix;
}


test("Staircase", () => {
    const n = 6;
    const result = staircase(n);
    for (let i = 0; i < n; i++) expect(result[0][i]).toEqual( (i >= 5) ? '#' : ' ');
	for (let i = 0; i < n; i++) expect(result[1][i]).toEqual( (i >= 4) ? '#' : ' ');
	for (let i = 0; i < n; i++) expect(result[5][i]).toEqual( (i >= 0) ? '#' : ' ');
});

File: java/src/hackerrank/Staircase.java
package hackerrank;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

//https://www.hackerrank.com/challenges/staircase/problem
public class Staircase {

	public static Character[][] staircase(int n) {
		if (n == 0) return null;
		Character[][] stairMatrix = new Character[n][n];
		for (int row = 0; row < n; row++) {
			for (int col = 0; col < n; col++) {
				stairMatrix[row][col] = col >= ( (n-row) - 1 ) ? '#' : ' ';
			}
		}
		return stairMatrix;
	}


	@Test
	void testStaircase() {
		int n = 6;
		Character[][] result = staircase(n);
		for (int i = 0; i < n; i++) assertEquals(result[0][i].toString(), String.valueOf( (i >= 5) ? '#' : ' ') );
		for (int i = 0; i < n; i++) assertEquals(result[1][i].toString(), String.valueOf( (i >= 4) ? '#' : ' ') );
		for (int i = 0; i < n; i++) assertEquals(result[5][i].toString(), String.valueOf( (i >= 0) ? '#' : ' ') );
	}

}
